### PR TITLE
Typescript is annoying 0

### PR DESCRIPTION
**Pull request by danbush**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -24,11 +24,11 @@
...
-  var row1: number = articleSetForSingleSource.slice(0, 1)
-  var row2: number = articleSetForSingleSource.slice(2, 3)
-  var row3: number = articleSetForSingleSource.slice(3, 4)
-  var row4: number = articleSetForSingleSource.slice(4, 5)
-  var row5: number = articleSetForSingleSource.slice(5, 6)
+  var row1: number = articleSetForSingleSource.slice(0, 1)[0]
+  var row2: number = articleSetForSingleSource.slice(2, 3)[0]
+  var row3: number = articleSetForSingleSource.slice(3, 4)[0]
+  var row4: number = articleSetForSingleSource.slice(4, 5)[0]
+  var row5: number = articleSetForSingleSource.slice(5, 6)[0]
```
Comment (stephenmparke) [inline src/App.vue:31]: Just setting an index of 0 for now to make it not an array?

File: src/components/MayHaveMissed.vue [hunk 2]
```diff
@@ -18,8 +18,11 @@
...
-      const result = await chocolateSauce(hashtagBuildTheList(articleNumber));
-      articleArray.value = result;
+      const listItem: string | Array<string> | undefined = hashtagBuildTheList(articleNumber);
+      if (typeof listItem === "string") {
+        const result = await chocolateSauce(listItem);
+        articleArray.value = result;
+      }
     } catch (err) {
       console.log(err);
     }
```
Comment (stephenmparke) [inline src/components/MayHaveMissed.vue:28]: Would be nice moving forward to also output a fun front-facing something like "WHOOPS! WHOOPSIES!" for a few reasons. would be nice to not have to dig through inspector, while also adding WHIMSY

File: src/components/MultipleHeadlines.vue [hunk 3]
```diff
@@ -12,7 +12,7 @@
   // Define props for the component
   const { count = 4, articleNumbers = [0,1,2,3] } = defineProps<{
     count: number,
-    articleNumbers: number
+    articleNumbers: number | number[]
```
Comment (stephenmparke) [inline src/components/MultipleHeadlines.vue:15]: Is this just saying "either a single number, or if maybe an array of numbers is returned?

File: src/components/MultipleHeadlines.vue [hunk 4]
```diff
@@ -21,17 +21,26 @@
...
+    results.forEach((result, index) => {
+      if (result !== null) {  // skip the default values
+        const articleNumber = articleSet[index];
+        articleArray.value[articleNumber] = result;
+      }
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
```
Comment (stephenmparke) [inline src/components/MultipleHeadlines.vue:42]: I'll stop saying this, but see above. user-facing whimsy.

File: src/components/helpers/chocolate_sauce.ts [hunk 5]
```diff
@@ -203,8 +203,8 @@ export async function chocolateSauce(url: string, item: number = 0, starter: num
 	var article: any = {};
 	var article_title: string = '';
 	var article_body: string = '';
-	var article_image: string = '';
-	var article_logo: string = '';
+	var article_image: any = '';
```
Comment (stephenmparke) [inline src/components/helpers/chocolate_sauce.ts:206]: In what case is this failing? I'm trying to think of a reason anything besides a string (or null/empty) would end up here to start with.

File: src/components/helpers/chocolate_sauce.ts [hunk 6]
```diff
@@ -217,11 +217,11 @@ export async function chocolateSauce(url: string, item: number = 0, starter: num
...
-			article_publisher = masto_data?.article_publisher
-			article_logo = masto_data?.article_logo
-			article_publisher_url = masto_data?.article_publisher_url
-			article_url = masto_data?.article_url
-			article_image = masto_data?.article_image
+			article_publisher = masto_data?.article_publisher ?? ''
+			article_logo = masto_data?.article_logo ?? ''
+			article_publisher_url = masto_data?.article_publisher_url ?? ''
+			article_url = masto_data?.article_url ?? ''
+			article_image = masto_data?.article_image ?? ''
```
Comment (stephenmparke) [inline src/components/helpers/chocolate_sauce.ts:224]: My ignorance: what does the ?? operator do in this context?

**Review by stephenmparke — APPROVED** [review]
A++ YOU'RE A WINNER! Looks good.

Sent back with mostly clarification questions and some requests for explainers/my education.

One suggested change, but nothing stopping this.

Nice.

File: src/App.vue (hunk 1 above)
Comment (danbush) [reply src/App.vue:31]: It’s more that waffle works with both multi and single cards. So the single cards expect a single number. Eventually I’ll probably just change the single cards to expect an array of one

File: src/components/MultipleHeadlines.vue (hunk 3 above)
Comment (danbush) [reply src/components/MultipleHeadlines.vue:15]: Yeah

File: src/components/helpers/chocolate_sauce.ts (hunk 5 above)
Comment (danbush) [reply src/components/helpers/chocolate_sauce.ts:206]: Sometimes the image pulled ends up an object. I forget why

**Pull request merged**